hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/PutObjectOptions.java

```diff
@@ -26,6 +26,16 @@
  */
 public final class PutObjectOptions {
 
+  /**
+   * Can the PUT operation skip marker deletion?
+   */
+  private final boolean keepMarkers;
+
+  /**
+   * Is this a conditional PUT operation
+   */
+  private final boolean conditionalPutEnabled;
```

Comment:
given there's now a etag match too, maybe we should give this a name, such as

`noObjectOverwrite`

we can then add a new one after `etagOverwrite` which will be a string